### PR TITLE
RCIAM-428_AUP_model_construct

### PR DESCRIPTION
**Pull request by ioigoume**
Construct AUP model and add in the state - `$state['rciamAttributes']['aup']`. 
Add COPerson ID in the state - `$state['rciamAttributes']['userId']`.

### CODE REVIEW

File: lib/Auth/Process/COmanageDbClient.php [hunk 1]
```diff
@@ -61,7 +61,7 @@ class sspmod_attrauthcomanage_Auth_Process_COmanageDbClient extends SimpleSAML_A
     private $blacklist = array();
 
     private $coId;
-    private $coTermsId = null;
+    private $coTermsAgreement = null;
```
Comment (NicolasLiampotis) [inline lib/Auth/Process/COmanageDbClient.php:64]: false

File: README.md [hunk 2]
```diff
@@ -67,7 +67,7 @@ The following authproc filter configuration options are supported:
     * `voGroupPrefix`: An array of group prefixes per (CO)mmunity to be used for the composition of the entitlements. Defaults to `urlencode($co_name) . ":group"`.
     * `coUserIdType`: A string that indicates the type of the identifier that the users have. Defaults to `epuid`.
     * `coOrgIdType`: An array containing the Identifier types under the user's Organizational Identities. Defaults to `array('epuid')`.
-    * `coTermsId`: An integer that indicates the ID of the Terms Aggreement signed by the user for quering the COmanage Registry. Defaults to `null`.
+    * `$coTermsAgreement`: A boolean value for controlling whether to construct `AUP` model from COmanage Registry. The model is then stored in the state - `$state['rciamAttributes']['aup']`. Defaults to `null`.
```
Comment (NicolasLiampotis) [inline README.md:70]: false

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]

File: lib/Auth/Process/COmanageDbClient.php (hunk 1 above)
Comment (ioigoume) [reply lib/Auth/Process/COmanageDbClient.php]: Done

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]
* Rename `coTermsAgreement` option to `retrieveAUP`
* Document `$state['rciamAttributes']['aup']` model
* Document `$state['rciamAttributes']['userId']` model
* Change `$state['rciamAttributes']['userId']` to store the COPersonID. We probably need to think of a better name, e.g. `$state['rciamAttributes']['registryUserId']` 
* Other parts of the (basic) COmanage profile could be stored in other dedicated rciamAttributes but this can be handled in a new PR

File: lib/Auth/Process/COmanageDbClient.php [hunk 3]
```diff
@@ -1006,6 +1075,9 @@ private function retrieveCOPersonData(&$state)
         } else {
             $basicInfo = $this->getBasicInfo($orgId);
         }
+        // XXX Add CO Person Id in the state
+        $state['rciamAttributes']['registryUserId'] = $basicInfo['id'];
+
```
Comment (NicolasLiampotis) [inline lib/Auth/Process/COmanageDbClient.php:1080]: Add note to the filter documentation in the README:
The filter adds the following attributes to the state information that SimpleSAMLphp stores in association with the request:
```
$state['rciamAttributes`] => array(
    'registryUserId` => // User's COPerson ID 
    'aup' => array( ...
)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [unrealesed]
+
+### Added
+- Add user's Terms and Conditions status in the state
```
Comment (NicolasLiampotis) [inline CHANGELOG.md:10]: - User's Terms and Conditions/Acceptable Use Policy (AUP) information in state information
- User's COPerson ID in state information

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]

File: lib/Auth/Process/COmanageDbClient.php [hunk 5]
```diff
@@ -33,6 +32,7 @@
  *            ),
  *            'urnNamespace' => 'urn:mace:example.eu',
  *            'urnAuthority' => 'example.eu',
+ *            'coTermsAgreement' => true,
```
[the remark this answers is not in the document]
Comment (ioigoume) [reply lib/Auth/Process/COmanageDbClient.php]: Done

File: README.md [hunk 6]
```diff
@@ -67,7 +67,7 @@ The following authproc filter configuration options are supported:
     * `voGroupPrefix`: An array of group prefixes per (CO)mmunity to be used for the composition of the entitlements. Defaults to `urlencode($co_name) . ":group"`.
     * `coUserIdType`: A string that indicates the type of the identifier that the users have. Defaults to `epuid`.
     * `coOrgIdType`: An array containing the Identifier types under the user's Organizational Identities. Defaults to `array('epuid')`.
-    * `coTermsId`: An integer that indicates the ID of the Terms Aggreement signed by the user for quering the COmanage Registry. Defaults to `null`.
+    * `retrieveAUP`: A boolean value for controlling whether to construct `AUP` model from COmanage Registry. The model is then stored in the state - `$state['rciamAttributes']['aup']`. Defaults to `false`.
```
[the remark this answers is not in the document]
Comment (ioigoume) [reply README.md]: Done

File: CHANGELOG.md (hunk 4 above)
Comment (ioigoume) [reply CHANGELOG.md]: Done

File: README.md [hunk 7]
```diff
@@ -133,6 +138,15 @@ $tag = preg_replace('/attrauthcomanage:/','yourthememodule:', $this->data['e'],
 replacing ```yourthememodule``` with the name of your theme module. 
 Also you must copy the ```attrauthcomanage.definition.json``` file under `yourthememodule/dictionaries` and then change the error messages in order to override the defaults.
 
+### State Attributes
+The filter adds the following attributes to the state information that SimpleSAMLphp stores in association with the request:
+```php
+$state['rciamAttributes'] => array(
+    'registryUserId' => // User's COPerson ID 
+    'aup' => array( ...
+)
```
Comment (NicolasLiampotis) [inline README.md:147]: this is incomplete. can you add the $coPersonId (int) and the actual aup model or a reference to models/AUP.md?

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]

**Review by NicolasLiampotis — APPROVED** [review]

**Pull request merged**